bughouse_console/src/async_server_main.rs

```diff
@@ -15,29 +15,47 @@ use rand::RngCore;
 use tide::StatusCode;
 use tungstenite::protocol;
 
-use bughouse_chess::server::*;
 use bughouse_chess::server_hooks::ServerHooks;
+use bughouse_chess::{server::*, BughouseServerEvent};
 use bughouse_console::auth;
 use bughouse_console::database;
 use bughouse_console::http_server_state::*;
-use bughouse_console::session::Session;
+use bughouse_console::session_store::*;
 
 use crate::auth_handlers_tide::*;
 use crate::network::{self, CommunicationError};
 use crate::server_main::{AuthOptions, DatabaseOptions, ServerConfig, SessionOptions};
 use crate::sqlx_server_hooks::*;
 
-async fn handle_connection<S: AsyncRead + AsyncWrite + Unpin + Send + Sync + 'static>(
+async fn handle_connection<DB, S: AsyncRead + AsyncWrite + Unpin + Send + Sync + 'static>(
     peer_addr: String,
     stream: WebSocketStream<S>,
     tx: mpsc::SyncSender<IncomingEvent>,
     clients: Arc<Mutex<Clients>>,
-    session: Option<Session>,
+    session_id: Option<SessionId>,
+    http_server_state: HttpServerState<DB>,
 ) -> tide::Result<()> {
     let (mut stream_tx, mut stream_rx) = stream.split();
-    info!("Client connected: {}, session={:?}", peer_addr, session);
+    info!("Client connected: {}", peer_addr);
 
     let (client_tx, client_rx) = mpsc::channel();
+
+    let mut session_store_subscription_id = None;
+    if let Some(session_id) = &session_id {
```

Comment:
Out of curiosity... Do you find
```
let session_store_subscription_id = session_id.map(...);
```
to be less readable?

(the benefit is, of course, reduced mutability)

Reply:
I don't have a preference really. I tend to do this (not always conciously) when the lambda is large-ish. Done with a map.